### PR TITLE
Deploy with yaml

### PR DESCRIPTION
**Pull request by nicksnyder**
This PR changes how we deploy Sourcegraph to Kubernetes.

The highlights are:
- Helm is neither required nor recommended.
- Base configuration is checked in as pure yaml (no Go templates).
- Test cases are not checked in to the repo (small diffs)

fixes https://github.com/sourcegraph/deploy-sourcegraph/issues/34
fixes https://github.com/sourcegraph/sourcegraph/issues/12229
fixes https://github.com/sourcegraph/sourcegraph/issues/10767
fixes 
https://github.com/sourcegraph/sourcegraph/issues/10803#issuecomment-414672919
fixes https://github.com/sourcegraph/sourcegraph/issues/12825
fixes https://github.com/sourcegraph/sourcegraph/issues/12809
fixes https://github.com/sourcegraph/sourcegraph/issues/12806

TODO
- [x] Address minor comments in https://github.com/sourcegraph/deploy-sourcegraph/pull/72
- [x] Link to tag/branch/commit of old deploy strategy from main README for existing customers
- [ ] Dogfood new process on fresh cluster
- [x] Update yaml tags to 2.11.0
- [x] Update README.dev.md for parallel release process

POST MERGE
- [ ] Update blog post https://docs.google.com/document/d/1xZUzqo-mae3vZX13uqP2RNG0SC7Z4DuaHHxSHWqxpy8/edit?ts=5b8895f6#heading=h.grx0305jtuiw

### CODE REVIEW

File: docs/scale.md [hunk 1]
```diff
@@ -59,81 +64,122 @@ table. See the following sections for how to modify your `values.yaml` file to a
...
-      symbols:
-        limits:
-          cpu: "4"
-          memory: 4G
-        requests:
-          cpu: "4"
-          memory: 4G
-  gitserver:
-    shards: 3
+# base/frontend/sourcegraph-frontend.Deployment.yaml
```
Comment (nicksnyder) [inline docs/scale.md]: I would delete these examples. It just makes it seem more complicated than it is. We don't need an example to update one field.

File: docs/scale.md [hunk 2]
```diff
@@ -59,81 +64,122 @@ table. See the following sections for how to modify your `values.yaml` file to a
...
-    infrastructure that does not support online volume resizing. To fix this, first `helm rollback
-    sourcegraph N` (where `N` is the previous version number as reported by `helm history sourcegraph`).
-    Then run `kubectl delete deployment indexed-search && kubectl delete pvc
-    indexed-search` and `helm upgrade sourcegraph ./helm-chart` again.
-*   If your change requires `gitserver` pods to be restarted and they are scheduled on another node
-    when they restart, they may go offline for 60-90 seconds (and temporarily show a `Multi-Attach`
-    error). This delay is caused by Kubernetes detaching and reattaching the volume. Mitigation
-    steps depend on your cloud provider; [contact us](https://about.sourcegraph.com/contact/) for
-    advice.
+~~_- After you change the `gitserver` shard count and run `helm upgrade`, you may see an error of the
```
Comment (nicksnyder) [inline docs/scale.md]: yes, delete

File: docs/scale.md [hunk 3]
```diff
@@ -272,125 +339,14 @@ Kubernetes cluster, based on the number of users in your organization:
 ## Using heterogeneous node pools with `nodeSelector`
```
Comment (nicksnyder) [inline docs/scale.md]: isn't this already documented in configure.md? (Assign resource-hungry pods to larger nodes)

Can we unify that doc?

File: configure/xlang/experimental/bash/xlang-bash.Deployment.yaml [hunk 4]
```diff
@@ -3,6 +3,9 @@ kind: Deployment
 metadata:
   annotations:
     description: LSP server for bash (used for live requests).
+  labels:
+    app: xlang-bash
```
Comment (nicksnyder) [inline configure/xlang/experimental/bash/xlang-bash.Deployment.yaml]: I am not sure if this is desirable. Services care about this, but it doesn't make sense for a service to be aware of a deployment? Probably doesn't do any harm, but maybe work checking

Comment (ggilmore) [reply configure/xlang/experimental/bash/xlang-bash.Deployment.yaml]: https://github.com/sourcegraph/deploy-sourcegraph/pull/58/commits/31c5cd762706253ae6e15ba61bb2fc06c5daf1c1

**Comment by keegancsmith** [comment]
> nicksnyder merged commit db21c90 into master  23 hours ago

🎉 🎉 🎉

**Pull request merged**